compiler/backend/src/org/jetbrains/kotlin/codegen/optimization/RedundantGotoMethodTransformer.kt

```diff
@@ -62,8 +62,7 @@ class RedundantGotoMethodTransformer : MethodTransformer() {
                         currentLabels.clear()
                     }
                 }
-                insn is LineNumberNode -> pendingGoto = null
-                insn.isMeaningful -> {
+                insn is LineNumberNode || (insn.isMeaningful && insn.opcode != Opcodes.NOP) -> {
```

Comment:
Could you clarify a bit this change?

Reply:
As the comment in JvmDefaultArgumentStubGenerator says, you cannot have a jump between the store into the argument and the "had explicit parameter" label. Problem is, the IR codegen may insert a bunch of NOPs, resulting in something like

```
    IFEQ Ln
    ...
    STORE n
    GOTO Ln
    NOP
 Ln
```

The NOP is removed by RedundantNopsCleanupMethodTransformer, but it only runs after RedundantGotoMethodTransformer (as otherwise it may decide LINENUMBER+NOP+GOTO can be shortened to just LINENUMBER+GOTO, but then removing the GOTO leaves no instruction to attach LINENUMBER to). Obviously, however, jumping over NOPs is unnecessary, so this change removes the GOTO even though there's a NOP between it and the target. (Clearing `currentLabels` on a line number is necessary to avoid jumping over LINENUMBER+NOP pairs.)

Reply:
Added a small comment.